File: MusicMaker/Core/Wave/sine_wave.cpp
//
//  sine_wave.cpp
//  MusicMaker
//
//  Created by Sergey Smagleev on 04.06.20.
//  Copyright © 2020 Sergey Smagleev. All rights reserved.
//

#include "sine_wave.hpp"
#include "math.h"

float CSineWave::shape(float value) {
    return sinf(value);
}

CSineWave::CSineWave(float _frequency, float _amplitude, float _phaseShift) : CWave(_frequency, _amplitude, _phaseShift) { }

CSineWave::~CSineWave() { }

File: MusicMaker/MusicEngine/SoundSynthesizer/Waves/SineWave.swift
//
//  SineWave.swift
//  MusicMaker
//
//  Created by Sergey Smagleev on 27.05.20.
//  Copyright © 2020 Sergey Smagleev. All rights reserved.
//

import Foundation

class SineWave: Wave {
    
    override func shape(_ value: Float) -> Float {
        sine_wave(value)
    }
    
}
